### PR TITLE
Adds Scan upsell when not available

### PR DESCRIPTION
**Pull request by ChaosExAnima**
#### Changes proposed in this Pull Request

* Adds a new scan page to promote Scan if it's not part of the site purchases.

#### Implementation notes

There's a few things that could be improved here:

* The detection on whether scan is enabled for a given site is entirely based around the site purchases.
* The purchases _only_ look for the Scan product, it doesn't check plans that would contain it.
* Enabling redirects to properly send users to a given destination to purchase Scan are needed.
* We still need to properly set the sidebar threat indicator, but I didn't want to bloat the PR, especially if we're improving the scan detection.

#### Testing instructions

* Load Jetpack Cloud.
* Select a site without scan purchased.
* Visit scan and scan history.
* Select a site with scan purchased.
* Visit scan and scan history.

Expected: sites without scan show new upsell screen, below. Sites with scan function as before.

Screen Shot 2020-04-08 at 4 28 24 PM

Fixes 1151678672052943-as-1170324772452809.

### CODE REVIEW

File: client/landing/jetpack-cloud/sections/scan/upsell.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	return (
+		<Main wideLayout className="scan__main">
+			<DocumentHead title="Scanner" />
+			<SidebarNavigation />
+			<div className="scan__content">
+				<SecurityIcon icon="info" />
+				<h1 className="scan__header">{ translate( 'Your site does not have scan' ) }</h1>
+				<p>
+					{ translate(
+						'Get automatic scanning and one-click fixes to keep your site one step ahead of security threats.'
```
Comment (a8ci18n) [inline client/landing/jetpack-cloud/sections/scan/upsell.jsx:30]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Automatic scanning and one-click fixes keep your site one step ahead of security threats.' )` *ES Score: 10*

**Comment by enejb** [comment]
Thanks for working on this @ChaosExAnima ! 

This tested pretty good for me but I found a couple of issues. 

When a site does have scan. We display the wrong icon on the scanner page see:
Screen Shot 2020-04-09 at 10 15 32 AM


For a site that does have a scan, because it has a plan we still tell the site that it needs scan. 
We should clarify that soon with Yamato how it should all work. 

I think it would be best to check that the site has a plan as well for now.

**Comment by ChaosExAnima** [comment]
Hey @enejb! I've updated things so that the correct icon is used again, and I now also check if the plan contains Scan as a feature.

File: client/landing/jetpack-cloud/sections/scan/controller.js [hunk 2]
```diff
@@ -2,22 +2,56 @@
...
+export function showUpsellIfNoScan( context, next ) {
+	const getState = context.store.getState;
+	const dispatch = context.store.dispatch;
+	const siteId = getSelectedSiteId( getState() );
+	let purchases = getSitePurchases( getState(), siteId );
+
+	const showUpsellOrNext = () => {
+		const hasScan = purchases.find(
+			purchase =>
+				JETPACK_SCAN_PRODUCTS.includes( purchase.productSlug ) ||
```
Comment (rcanepa) [inline client/landing/jetpack-cloud/sections/scan/controller.js:29]: I think I have seeing the same check in other places – checking if a user has a product by looking at purchases and plans. Do you think creating a selector would be a good idea? I'm referring to this piece of code:
```js
JETPACK_SCAN_PRODUCTS.includes( purchase.productSlug ) ||
( getPlan( purchase.productSlug ) &&
	planHasFeature( purchase.productSlug, PRODUCT_JETPACK_SCAN ) )
```

Comment (ChaosExAnima) [reply client/landing/jetpack-cloud/sections/scan/controller.js:29]: @rcanepa - We're discussing creating a specific API endpoint to query what the current state of scan is for a site, so I feel it's not worth the time to create a selector that queries purchases like this. Once we have the API in place we can deprecate this approach in favor of the new selector.

Comment (rcanepa) [reply client/landing/jetpack-cloud/sections/scan/controller.js:29]: Cool. That makes much more sense!

**Pull request merged**